deepchem/feat/molecule_featurizers/mat_featurizer.py

```diff
@@ -0,0 +1,79 @@
+from deepchem.feat.base_classes import MolecularFeaturizer
+from deepchem.utils.molecule_feature_utils import one_hot_encode
+try:
+  import logging
+  import numpy as np
+  import pandas as pd
+  import torch
+  from rdkit import Chem
+  from rdkit.Chem import AllChem
+  from rdkit.Chem import MolFromSmiles
+  from sklearn.metrics import pairwise_distances
+  from torch.utils import datasets
+except:
+  raise ImportError("Required Modules not found.")
+
+
+class MATFeaturizer(MolecularFeaturizer):
+
+  def __init__(
+      self,
+      add_dummy_node=True,
+      one_hot_formal_charge=True,
+  ):
+    self.mol = mol
+    self.add_dummy_node = add_dummy_node
+    self.one_hot_formal_charge = one_hot_formal_charge
+
+  def atom_features(self, atom, one_hot_formal_charge=True):
```

Comment:
DeepChem already has `atom_features` inside `deepchem/feat/graph_features.py`. Add a docstring saying that you use MAT specific features here. Also, write a docstring and test for everything as you go along, and not after you have written the full function. I understand things change, but it is good practice to write them parallely, and one needs to make a habit out of it.

Reply:
Thank you, I will keep this in mind and also add the docstring/s.